debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/events/BufferingChangeStreamCursor.java

```diff
@@ -297,8 +307,18 @@ public BufferingChangeStreamCursor<TResult> start() {
     @Override
     public ResumableChangeStreamEvent<TResult> tryNext() {
         var event = pollWithDelay();
-        if (event != null) {
-            lastResumeToken = event.resumeToken;
+        try {
+            if (event != null) {
+                lastResumeToken = event.resumeToken;
+            }
+            else {
+                if (!fetcher.isRunning() && fetcher.isEmpty()) {
+                    throw new MongoChangeStreamException("Failed to fetch event from change stream cursor, cursor is closed");
+                }
+            }
+        }
+        catch (MongoException e) {
+            throw new MongoException(e.getMessage());
```

Comment:
Please add original exception to the stack

Reply:
Definitively Could use DebeziumException as a wrapper